rfcs/react-components/convergence/event-handlers-event-type.md

```diff
@@ -32,71 +32,76 @@ We cannot just add the `Event` type to the existing callbacks as this is a break
 
 Use generic [`Event`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/v17/global.d.ts#L10) type and strongly-typed events in "data bag".
 
+### Implementation
+
+Define two helper types `EventData` and `EventHandler` to ensure consistency across all components:
+
+```ts
+export type EventData<Type extends string, TEvent> =
+  | { type: undefined; event: React.SyntheticEvent | Event | undefined }
+  | { type: Type; event: TEvent };
+
+export type EventHandler<TData extends EventData<string, unknown>> = (
+  ev: React.SyntheticEvent | Event | undefined,
```

Comment:
Same thing here. Why do we need undefined here? there's not a single case where we will not be providing an event, right?! I believe a generic signature without undefined suffices for this case
```suggestion
  ev: React.SyntheticEvent | Event,
```

Reply:
So we found out Tooltip is the only one so far that uses `undefined` event in callback https://github.com/microsoft/fluentui/blob/3129d7868082c057cc01ad860a45a9ffc8072a7b/packages/react-components/react-tooltip/src/components/Tooltip/Tooltip.types.ts#L67

@behowell WDYT if we remove `undefined` from `EventHandler` type, and for Tooltip `onVisibleChange`, we don't use `EventHandler` helper